### ISSUE TITLE
Refactor `AdSenseAlerts` to use the new Notifications datastore

### ISSUE THREAD

**Issue by zutigrm** [google/site-kit-wp#9300]
## Feature Description

This issue should refactor the `AdSenseAlerts` so that it uses the new datastore infrastructure to register and queue the notification. It should also incorporate a new "layout".

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The `AdSenseAlerts` component should be refactored so that AdSense notifications are added using the new `core/notifications` datastore—eg. the updated `registerNotification()` action that adds notifications to the queue since #9453.
  * Any notifications fetched should be registered using `registerNotification()` in the `NOTIFICATION_AREAS.HEADER` area, using data provided by the server notifications.

## Implementation Brief

* [ ] Each notification in the `notifications` variable (from `select( MODULES_ADSENSE ).getNotifications()`) should be added to the notifications data store using the `registerNotification()`. This will cause the notification to be added to the relevant queue. This can be done with a `useEffect`/similar.
  * [ ] Each notification is from the service and is "generic", we currently create each notification in a `notifications.map` section: https://github.com/google/site-kit-wp/blob/085243e0d197ff925e97c7301c27f415931e8a7b/assets/js/components/notifications/CoreSiteBannerNotifications.js#L84-L100. We'll want to map as much of the props from the server notifications to `registerNotification` as possible.
  * [ ] The `Component` used should be a generic component (similar to/the same as `ServerNotification` in `assets/js/components/ServerNotification.js` proposed in #9294) that accepts extra props like the `ctaLabel`, `learnMoreLabel`, etc. and uses the generic `<Notification>` component to render that notification with the proper CTA/dismiss/learn more buttons/links.
    * [ ] It would be wise to rename this component to be a bit more generic to allow server notifications and AdSense notifications to use it.
  * [ ] While `registerNotification()` already prevents duplicate notifications from being registered/re-registered, it would be best to prevent notifications from being registered again by keeping track of which AdSense notifications have already been registered.

### Test Coverage

* Tests should mostly be removed for functions that aren't needed anymore.
* Tests that check if AdSense notifications appear in a queue could be added.

## QA Brief

Because it's not possible to trigger server notifications reliably, the notifications need to be manually added to the server notification array.

* Use any site running Site Kit with Site Kit setup.
* Ensure the AdSense module is set up
* Insert the following code into a browser console to simulate a server notification:

```js
googlesitekit.data.dispatch("modules/adsense").receiveGetNotifications(
  [
    {
      id: "test-ad-notification",
      title: "AdSense Beta",
      content: "New version of AdSense!",
      ctaURL: "https://adsense.google.com/start/",
      ctaLabel: "Upgrade now!",
      ctaTarget: "_blank",
      learnMoreURL:
        "https://www.wordhippo.com/what-is/another-word-for/learn_more.html",
      learnMoreLabel: "Learn more",
      dismissible: true,
      dismissLabel: "Dismiss this message",
    },
  ],
  {}
);
```

You should observe a sever notification with buttons that dismiss the notification when clicked.

* Set up a new site with Site Kit installed, but do NOT set up the AdSense module.
* Copy the same code from above into the console and run it.
* No notification should appear, because the AdSense module is not connected.

## Changelog entry

* N/A

**Comment by eugene-manuilov** [comment]
AC ✔️

**Comment by zutigrm** [comment]
I am un-assigning myself since I have been caught up with higher priority epic issues, and will be on the PTO for few days. This notification will take a bit of a thinking it over, as it is potentially a group of notifications that are fetched from the backend, and current notifications API might need some tweaks to support this. The idea I had when initially had time to check this one, is that regular approach would potentially work, as we would register the notification in the API, and then `checkRequirements` and notification component would handle the fetching and iteration over data. Something to consider was that `isDismissable` is a fixed property included when notification is registered in the API, if the notifications coming from the backend are different in that regard, it won't render correctly, since `isDismissable` would be applied to all with the same value either `true` or `false`

**Comment by benbowler** [comment]
I'm marking this blocked by #9294 as there is an ongoing discussion there regarding the module level `createNotificationsStore`. This in an example of notifications that use this existing infrastructure so we need a unified spec/decision on how we want to approach this existing notifications store to bring it into the new store.

**Comment by benbowler** [comment]
Note around the priority of these banners, AdSense banners are the only banners which should be higher priority than ModuleRecoveryAlert which I have set in #10398, so in this ticket we should set it to higher priority (lower number) and other refactored BannerNotifications should be lower priority (higher number).

**Comment by jimmymadon** [comment]
Having discussed this and #9294 with @aaemnnosttv, we decided to hold off on these issues to prevent "nested" queues of notifications. This is because these notification components call a `getNotifications` selector which fetches one or more notifications and maps through them to render them. We might be better off creating these as "on-demand" notifications and so, we will wait till #9453 is completed first before we amend the AC/IB here.

**Comment by eugene-manuilov** [comment]
IB ✔

**Comment by kelvinballoo** [comment]
### QA Update ⚠️ 

Tested this and it's looking good overall.
Just a question:

**ITEM 1:**
The QAB states: "You should observe a sever notification with buttons that dismiss the notification when clicked"

We actually get 2 buttons:
- Upgrade now!
- Dismiss this message

Clicking 'Dismiss this message' will dismiss it accordingly with the relevant GA tracking.
When clicking on 'Upgrade now!', this would not dismiss it but will send us to the Google adsense page. I assume that's expected.  Can you confirm?
That said, it will open it in the same tab. Is that expected or it should be in a new tab?

<Details>

https://github.com/user-attachments/assets/f4a64d7e-52bb-4e33-b14d-714b270c1249

Clicking on 'Upgrade'


</Details>
________________________________________________________________________________

Other than that, the gist of the testing is fine:
- When Adsense is connected, the server notification appears and it's able to be dismissed. GA event being fired accordingly.
   <Details>
   
https://github.com/user-attachments/assets/aa878063-bae0-46d1-a805-f98883d7128f

   </Details>

- When Adsense is not connected or if it was not ever installed, no server notification would appear.
   <Details>
   
   https://github.com/user-attachments/assets/548bf5e1-b4f6-4c7c-987a-78515847e59d

   </Details>

**Comment by tofumatt** [comment]
> When clicking on 'Upgrade now!', this would not dismiss it but will send us to the Google AdSense page. I assume that's expected. Can you confirm?

Yes, that's expected behaviour with the test notification 👍🏻 

> That said, it will open it in the same tab. Is that expected or it should be in a new tab?

That's correct, it looks like these CTA handlers don't currently support target properties, I mistakenly added it as part of the data but it goes unused.

That's fine though, so this can be considered all good 🙂

**Comment by wpdarren** [comment]
### QA Update: ❌ 

@tofumatt @jimmymadon I am seeing this regression on my live site, where the AdSense alert content is not appearing in the banner.  You can see the issue in the screenshot below.

<img width="3270" height="786" alt="Image" src="https://github.com/user-attachments/assets/ca414ceb-a355-4b7d-866d-7441bf532304" />

**Comment by hussain-t** [comment]
Hi @wpdarren, the follow-up [PR](https://github.com/google/site-kit-wp/pull/11184) from @tofumatt has been merged. It’s now ready for another round of QA. Thanks!

**Comment by wpdarren** [comment]
### QA Update: ✅ 

**Verified:**

- When AdSense is enabled the alert notifications appeared on the Site Kit main dashboard. 

<img width="3266" height="832" alt="Image" src="https://github.com/user-attachments/assets/d793be3c-802b-4c06-a862-a3c170013e4e" />

**Note:** there is an odd UX experience which is inconsitent with other banners, so I will flag this with Jimmy separately, but for this bug, we have fixed it and the alert now appears.
